_includes/rest/objects.md

```diff
@@ -99,6 +99,8 @@ The response body is a JSON object containing the `objectId` and the `createdAt`
 }
 ```
 
+The response body also includes any field that was modified using `beforeSave` and `afterSave` triggers.
```

Comment:
Not sure about `afterSave`, the response is returned before the after save runs, right?

Reply:
No. The response is returned after `afterSave` trigger. And in `afterSave` trigger you can change the fields that will be returned. Take a look in this test here: https://github.com/parse-community/parse-server/blob/master/spec/CloudCode.spec.js#L187

Maybe it is not well written to explain this case (the one in the test). Any suggestion?

Reply:
That test is for the `beforeSave` trigger?

Looking in the cloud code guide under [Async Behaviour](http://docs.parseplatform.org/cloudcode/guide/#async-behavior) is what gives me this impression. Is it the case that if a promise is returned or async/await is used the response is returned after the `afterSave` completes but otherwise the response is returned before the `afterSave` completes?

Sorry if I'm not making sense, I'm just migrating my cloud code to 3.0.0+ and my knowledge of promises and async/await is limited.

Reply:
If the `afterSave` trigger returns a Promise, Parse Server will wait this promise to resolve before returning to the client and you can even change the fields that will be returned (as showed in the test case that I sent you before). If the `afterSave` trigger does not return a promise (as in the example of the docs that you sent me), the response will be sent to the client and the trigger may continue running.

Reply:
I see, thanks for bringing me up to speed on that, I might work on marking the docs more clear on that point.